files/brews/nginx.rb

```diff
@@ -1,67 +1,132 @@
 require 'formula'
 
 class Nginx < Formula
-  homepage 'https://nginx.org/'
-  url 'https://nginx.org/download/nginx-1.10.2.tar.gz'
-  sha256 '1045ac4987a396e2fa5d0011daf8987b612dd2f05181b67507da68cbe7d765c2'
-  version '1.10.2-boxen1'
+  desc "HTTP(S) server and reverse proxy, and IMAP/POP3 proxy server"
+  homepage "https://nginx.org/"
+  url "https://nginx.org/download/nginx-1.12.1.tar.gz"
+  sha256 "8793bf426485a30f91021b6b945a9fd8a84d87d17b566562c3797aba8fac76fb"
+  version '1.12.1-boxen1'
 
-  depends_on 'pcre'
+  bottle do
+    sha256 "93bcf8e3aec465c219b6c0b4f4d5437c61bf00f2a930ef5702e0521edc51f20e" => :sierra
+    sha256 "8a7c3580534aa0854927f750d4f044a2a85f90d4c1936338a4a09fef7db0824e" => :el_capitan
+    sha256 "0caae754f402abbe1eca413a7f0291fe2499d5779bb1e537d7f80a4d7d3156d3" => :yosemite
+  end
+
+  # Before submitting more options to this formula please check they aren't
+  # already in Homebrew/homebrew-nginx/nginx-full:
+  # https://github.com/Homebrew/homebrew-nginx/blob/master/Formula/nginx-full.rb
+  option "with-passenger", "Compile with support for Phusion Passenger module"
+  option "with-webdav", "Compile with support for WebDAV module"
+  option "with-debug", "Compile with support for debug log"
+  option "with-gunzip", "Compile with support for gunzip module"
+
+  depends_on "pcre"
+  depends_on "passenger" => :optional
 
-  skip_clean 'logs'
+  # passenger uses apr, which uses openssl, so need to keep
+  # crypto library choice consistent throughout the tree.
+  if build.with? "passenger"
+    depends_on "openssl"
+  else
+    depends_on "openssl@1.1"
+  end
 
-  def options
-    [
-      ['--with-passenger',   "Compile with support for Phusion Passenger module"],
-      ['--with-webdav',      "Compile with support for WebDAV module"],
-      ['--with-gzip-static', "Compile with support for Gzip Static module"],
-      ['--with-http2',       "Compile with support for the HTTP/2 module"],
+  def install
+    pcre = Formula["pcre"]
+
+    if build.with? "passenger"
+      openssl = Formula["openssl"]
+    else
+      openssl = Formula["openssl@1.1"]
+    end
+
+    cc_opt = "-I#{pcre.opt_include} -I#{openssl.opt_include}"
+    ld_opt = "-L#{pcre.opt_lib} -L#{openssl.opt_lib}"
+
+    args = %W[
+      --prefix=#{prefix}
+      --with-http_ssl_module
+      --with-pcre
+      --sbin-path=#{bin}/nginx
+      --with-cc-opt=#{cc_opt}
+      --with-ld-opt=#{ld_opt}
+      --conf-path=#{etc}/nginx/nginx.conf
```

Comment:
I've opened a discussion issue at boxen/boxen#214 since the configuration for Homebrew cannot be outside of the `/etc` directory anymore.

I'm looking into changing this across the board for all puppet modules (nginx, mysql, dnsmaq, etc) but not quite sure on the rollout plan for that just yet but I will open some PRs.